Seminar-IS/Seminar02/Pract 02/task03.cpp

```diff
@@ -0,0 +1,157 @@
+#include <iostream>
+#include <fstream>
+#include <string.h>
+
+using namespace std;
+
+#pragma warning(disable : 4996)
+
+enum class Tyres {
+    Soft=45,
+    Medium=50,
+    Hard=60,
+    Intern=40,
+    Wet=35
+};
+
+const size_t MAX_NAME_LENGTH = 65;
+const size_t MAX_TYRE_LENGTH = 10;
+const size_t MAX_ROW_SIZE = 70;
+
+struct Bolid {
+    double tyreWearCoefficient;
+    size_t maxLoops;
+    Tyres tyres;
+    short int yearOfManafacture;
+    char name[MAX_NAME_LENGTH];
+    bool inRepairStation;
+};
+
+void readTyres(Tyres& tyres) {
+    char* type = new char[MAX_TYRE_LENGTH];
+    cin.getline(type, MAX_TYRE_LENGTH);
+    if (strcmp(type, "Soft") == 0) {
+        tyres = Tyres::Soft;
+    }
+    else if (strcmp(type, "Medium") == 0) {
+        tyres = Tyres::Medium;
+    }
+    else if (strcmp(type, "Hard") == 0) {
+        tyres = Tyres::Hard;
+    }
+    else if (strcmp(type, "Intern") == 0) {
+        tyres = Tyres::Intern;
+    }
+    else if (strcmp(type, "Wet") == 0) {
+        tyres = Tyres::Wet;
+    }
+    delete[] type;
+}
+
+void readBolid(Bolid& bolid) {
+    cout << "Name: ";
+    cin.getline(bolid.name, MAX_NAME_LENGTH);
+    cout << "Year of manafacture: ";
+    cin >> bolid.yearOfManafacture;
+    cin.ignore();
+    cout << "Tyres: ";
+    readTyres(bolid.tyres);
+    cout << "Tyre wear coefficient: ";
+    cin >> bolid.tyreWearCoefficient;
+    cout << "Is in repair station?: ";
+    bolid.inRepairStation = cin.get()=='1';
+    cin.ignore();
+    cout << "Max loops: ";
+    cin >> bolid.maxLoops;
+    cin.ignore();
+}
+
+size_t calcMaxLoops(const Bolid& bolid) {
+    return bolid.maxLoops * bolid.tyreWearCoefficient;
+}
+
+void sortBolids(Bolid* bolids, const size_t size) {
+    size_t* indexes = new size_t[size];
+    for (size_t i = 0; i < size; i++) {
+        indexes[i] = i;
```

Comment:
?

Reply:
That was some old code which I haven't removed.